### ISSUE TITLE
Possible to customize the placeholder text?

### ISSUE THREAD

**Issue by steven-tey** [TypeCellOS/BlockNote#122]
Incredible work on BlockNote, Yousef!

For [Platforms Starter Kit](https://github.com/vercel/platforms) 2.0, I'm planning to integrate OpenAI text completions to the text editor.

Therefore, I was wondering if there's any way we can customize the placeholder text in BlockNote? My idea is to have a debounced callback to OpenAI's ChatGPT API to do text completions in the style of a placeholder text 😄 

LMK what you think! I'm happy to contribute to the source code as well if this feature is not available yet!

**Comment by YousefED** [comment]
Thanks @steven-tey! Happily running BlockNote on Vercel ❤️ 

Making it easy to build AI powered text editors is an important part of our vision. So your use-case fits right in, would be excited to work on this.

I'll see how easy we can expose placeholders and keep you posted.

Thinking along; to make this work with ok performance, it'd be best to fetch multiple completions from the API at the same time, right? So that when the user types something slightly different than the "main suggestion", there's still a chance we can show a different suggestion without having to wait for a new API call. I think this is how Codex suggestions work for example.

We'd also need to figure out at what level we want to expose the API; e.g. low-level `block.setPlaceholderText`, a bit higher level `block.setSuggestions()` or callback-style `block.registerSuggestionProvider()`. I'll give this some thought as well

**Comment by steven-tey** [comment]
Thanks for responding Yousef! I'm currently blocked by #132 so I'll get back to your questions above once I figure that out! 😄

**Comment by steven-tey** [comment]
@YousefED Update on this – I tried building this out in TipTap but it didn't seem like the most logical move from a UX standpoint 😅 I'm planning to just rely on Slash commands / Formatting toolbar to integrate AI features so I'm closing this for now!

**Comment by YousefED** [comment]
> @YousefED Update on this – I tried building this out in TipTap but it didn't seem like the most logical move from a UX standpoint 😅 I'm planning to just rely on Slash commands / Formatting toolbar to integrate AI features so I'm closing this for now! (feel free to reopen it if you think it's appropriate though)

@steven-tey could you tell me more about this? What's not ideal from UX perspective? I think allowing for something like Codex in VS Code would be pretty interesting - so was still planning to put this on the roadmap. Would be nice to hear your learnings so we can reconsider!

**Comment by YousefED** [comment]
@simon-lund makes sense, maybe we should open a separate issue for this - and maybe we can use that to brainstorm further requirements for an AI enabled editor there?

**Comment by devAbdullahAli77** [comment]
Can someone please guide me about how can i change the placholders for my custom components ? the official placeholder example is the docs is having issues. It gives error on importing { en } from @blocknote/core/locales. 

I am getting this error:

Cannot find module '@blocknote/core/locales' or its corresponding type declarations.
  There are types at 'c:/Users/E14/client/node_modules/@blocknote/core/types/src/i18n/index.d.ts', but this result could not be resolved under your current 'moduleResolution' setting. Consider updating to 'node16', 'nodenext', or 'bundler'

and on changing the moduleResolution, to the suggested ones, I get other errors! 
For now I copied the en file from github repo and placed it in my project and thats how I am able to use it, I even have defined placeholders for my custom components, but not working! However changing the placeholders for default components is working fine!

**Comment by nperez0111** [comment]
@devAbdullahAli77 to fix the issue you are running into, you do need to change the moduleResolution setting for the locales import to work properly. So, you'll have to fix the errors that come after changing the moduleResolution setting, this is something to do with your project, so I cannot be of any help with this.

See this example for changing placeholder text: https://www.blocknotejs.org/examples/basic/custom-placeholder

**Comment by devAbdullahAli77** [comment]
@nperez0111 Okay, thanks for your reply. I will update module resolution and do further changes accordingly.
